src/provider/provider_level_zero.c

```diff
@@ -181,6 +214,15 @@ static umf_result_t ze_memory_provider_alloc(void *provider, size_t size,
         return UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC;
     }
 
+    for (uint32_t i = 0; i < ze_provider->resident_device_count; i++) {
```

Comment:
I'm not sure but is calling zeContextMakeMemoryResident valid also for HOST allocations?

Reply:
Yes, UR makes the allocation resident on all devices in the context in such case. My idea here was that the user (UR) would pass an appropriate list of devices depending on the memory type.